pages/04.tutorials/32.confiugre-forbidden-sysctls/default.en.md

```diff
@@ -0,0 +1,163 @@
+---
+title: 'How to reconfigure forbidden sysctls'
+published: true
+date: '07-01-2021 21:30'
+taxonomy:
+    tag:
+        - metakube
+        - cluster
+        - sysctl
+        - kubelet
+        - security group
+        - pod security policy
+---
+
+At some point during your Kubernetes journey you will come to the point where
+you need to adjust some parameters of the underlying node system, either
+globally or in the context of a specific pod or set of pods. Maybe you are
+running Redis and need to
+[increase the amount of allowed maximum connections](https://github.com/redis/redis/blob/unstable/redis.conf#L100)
+on a socket or you use Apache webserver and want to
+[enable TCP Keepalive](http://httpd.apache.org/docs/current/mod/core.html#keepalive)
+but with different default parameters than the kernel compiled ones.
+
+To do so you need to adjust the according sysctl values. By default your
+Kubernetes installation
+[will prevent you from doing so](https://kubernetes.io/docs/tasks/administer-cluster/sysctl-cluster/)
+to ensure a secure and stable workload environment. If you try anyway e.g., by specifying:
+
+```yaml
+securityContext:
+  sysctls:
+  - name: "net.ipv4.tcp_keepalive_time"
+    value: "45"
+  - name: "net.ipv4.tcp_keepalive_intvl"
+    value: "45"
+  - name: "net.ipv4.tcp_keepalive_probes"
+    value: "10"
+```
+
+chances are you will see something like this in the event log output for your
+pods:
+
+```bash
+# kubectl describe pod/test
```

Comment:
Remove the `#` in the begining of the line to show it as a command and not a comment in the code block.